test/Microsoft.Azure.Devices.Edge.Test/Provisioning.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Test.Common;
+    using Microsoft.Azure.Devices.Edge.Test.Helpers;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using NUnit.Framework;
+
+    public class Provisioning : DeviceProvisioningFixture
+    {
+        protected readonly IotHub iotHub;
+
+        public Provisioning()
+        {
+            this.iotHub = new IotHub(
+                Context.Current.ConnectionString,
+                Context.Current.EventHubEndpoint,
+                Context.Current.Proxy);
+        }
+
+        string DeriveDeviceKey(byte[] groupKey, string registrationId)
+        {
+            using (var hmac = new HMACSHA256(groupKey))
+            {
+                return Convert.ToBase64String(hmac.ComputeHash(Encoding.UTF8.GetBytes(registrationId)));
+            }
+        }
+
+        [Test]
+        public async Task DpsSymmetricKey()
+        {
+            string idScope = Context.Current.DpsIdScope.Expect(() => new ArgumentException());
+            string groupKey = Context.Current.DpsGroupKey.Expect(() => new ArgumentException());
+            string registrationId = $"{Context.Current.DeviceId}-{TestContext.CurrentContext.Test.NormalizedName()}";
+
+            string deviceKey = this.DeriveDeviceKey(Convert.FromBase64String(groupKey), registrationId);
+
+            CancellationToken token = this.cts.Token;
+
+            await this.daemon.ConfigureAsync(
+                config =>
+                {
+                    config.SetDpsSymmetricKey(idScope, registrationId, deviceKey);
+                    config.Update();
+                    return Task.FromResult((
+                        "with DPS symmetric key attestation for '{Identity}'",
```

Comment:
Yes that is possible. ActiveState=active just means the iotedged process is running, and iotedged will run long enough to learn that DPS provisioning failed, then it will exit. In that case, this test may get past `this.daemon.WaitForStatusAsync(EdgeDaemonStatus.Running, token)` below, but then `agent.WaitForStatusAsync(EdgeModuleStatus.Running, token)` will time out.